### PR TITLE
PXP-662 fix: pipeline status request does not need builds

### PR DESCRIPTION
**Pull request by Fadhil**
## Summary

* we introduce a new query for status and last deployed build artifact, fixing a bug where the status for a build on a pipeline always fails when it is on github actions but doesn't have a jenkins pipeline.

Ticket:

## What's Changed

Added a `cdPipelineStatus` query

Changed `fetch_cd_pipeline_status` function to use said query

Fixes an issue where calls with `--canary` fail because Jenkins pipeline is not defined

#### Caveat
Wukong CLI clients that do not update to this fix will still be able to use the client as expected, but calls with `--canary` will fail. Updating the client to the new one with this fix will allow the client to work in all cases.

### CODE REVIEW

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/184#" title="Author self-approved">Fadhil</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by onimsha — APPROVED** [review]

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/184#" title="Author self-approved">Fadhil</a>*, *<a href="https://github.com/mindvalley/wukong-cli/pull/184#pullrequestreview-1830948973" title="Approved">onimsha</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
